### ISSUE TITLE
[Request] Treat curly and round parentheses separately

### ISSUE THREAD

**Issue by mkrnic** [tomasr/viasfora#151]
Hi :)

Could you please add an option to treat parentheses separately?

This is kinda too much:

![](http://i.imgur.com/CcKBfbp.png)

`Baz()` is green because it's inside pink curly, call to `quux()` is purple because it's inside green curly...

What I'd like is to disable coloring curly parentheses entirely, and basically keep Viasfora working just for nested round parentheses to help with `if (((foo and bar) or baz) or quux)` scenarios.

If scope could be inferred somehow, then I'd keep colored curly parentheses *only* inside functions and properties. Namespaces and classes, nope, I really don't need that.

But regardless of scope, curly and round should be separate.

**Comment by tomasr** [comment]
So what happens to square brackets (```[]```)? Are those on a different regime as well? ignored?

Highlighting curly braces only in some cases based on syntactic analysis is probably out the question, though, since Viasfora relies strictly on lexical analysis for this :disappointed:

**Comment by mkrnic** [comment]
I haven't tried square brackets :) But if they're also coloured by Viasfora, yeah - an option to treat them separately, too! You get an option, you get an option, everyone gets an option! oprah.jpg

I'm just basically looking for something to help with nested round parentheses, because various `if`s are often hard to figure out. I have no problems with curly/square coloring being left out, as I don't have code that looks like this in a single line `{{{x}y}z}` :)

**Comment by Falconne** [comment]
Yes, like in the Vim plugin, it would be good if curly brace colouring could be turned off separately to the parenthesis. I find braces being coloured distracting and unnecessary as scoped blocks of code are already indented anyway... maybe it would be useful for languages like PowerShell where curly braces are more heavily used.

It's also distracting that the parenthesis colouring is interleaved with the brace colouring, resulting in way too many colours on screen. I think it's easier for comprehension when parenthesis are coloured consistently based on their own nesting, without considering braces.
